src/inmanta/plugins.py

```diff
@@ -280,12 +277,25 @@ def to_type(self, arg_type: Optional[object], resolver):
         if arg_type == "expression":
             return None
 
-        if arg_type.endswith("[]"):
-            locatable_type.value = arg_type[0:-2]
-            basetype = resolver.get_type(locatable_type)
-            return TypedList(basetype)
+        filename: Optional[str] = inspect.getsourcefile(self.__class__.__function__)
+        location: Range
+        assert filename is not None
+        line: int = inspect.getsourcelines(self.__class__.__function__)[1] + 1
+        location = Range(filename, line, 1, line, 2)
+        locatable_type: LocatableString = LocatableString(arg_type, location, 0, None)
+
+        # transformations in reverse order
+        transformations: List[Callable[[InmantaType], InmantaType]] = []
```

Comment:
This next section of code is quite beautiful. It is also quite esoteric.

There should be a comment what the purpose of the transformations array is, so that it is understandable to the reader.